src/common/cockpitconf.c

```diff
@@ -212,6 +214,38 @@ load_key_file (const char *file_path)
   return ret;
 }
 
+/* Logic to load all additional .conf files from cockpit.conf.d directory */
+static void
+load_conf_d_files (void)
+{
+  const char *conf_d_dir = "/etc/cockpit/cockpit.conf.d";
+  DIR *dir;
+  struct dirent *entry;
+  struct stat st;
+  char filepath[PATH_MAX];
+
+  dir = opendir (conf_d_dir);
+  if (!dir)
+    return;
+
+  while ((entry = readdir (dir)) != NULL)
+    {
+      if (entry->d_type != DT_REG && entry->d_type != DT_UNKNOWN)
+        continue;
+
+      if (!g_str_has_suffix (entry->d_name, ".conf"))
+        continue;
+
+      int len = snprintf(filepath, sizeof(filepath), "%s/%s", conf_d_dir, entry->d_name);
+      if (len < 0 || (size_t)len >= sizeof(filepath))
+        continue;
+      if (stat(filepath, &st) == 0 && S_ISREG (st.st_mode))
+        load_key_file(filepath);
```

Comment:
I assume dropin files should act as "override" of the base cockpit.conf and looking at `cockpit_conf_init` we do support XDG_CONFIG_DIR which calls load_key multiple times.